pulser/waveforms.py

```diff
@@ -233,6 +269,39 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f'CompositeWaveform({self.duration} ns, {self._waveforms!r})'
 
+    def __getitem__(self, index_or_slice:
+                    Union[int, slice]) -> Union[float, np.ndarray]:
+        wf_start: int
+        if isinstance(index_or_slice, slice):
+            s: slice = self._check_slice(index_or_slice)
+            start: int = s.start
+            stop: int = s.stop
+            array: np.ndarray = np.array([])
+            wf_start = 0
+            for wf in self._waveforms:
+                if wf_start <= start and start <= wf_start+wf.duration:
+                    if wf_start <= stop and stop <= wf_start+wf.duration:
```

Comment:
You can write this as: 
```suggestion
                    if wf_start <= stop <= wf_start+wf.duration:
```
I feel like it's easier to read, don't you?

Reply:
> You can write this as:
> 
> I feel like it's easier to read, don't you?

Indeed. I didn't know this syntax.